daemon/inertiad/webhook/bitbucket_push.go

```diff
@@ -0,0 +1,59 @@
+package webhook
+
+import "strings"
+
+// bitbucketPushEvent represents a push to a Bitbucket repository
+// see https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html
+type bitbucketPushEvent struct {
+	eventType  string
+	branchName string
+	fullName   string
+}
+
+// Due to heavy nesting, extracting keys with type assertions is preferred
+func parseBitbucketPushEvent(rawJSON map[string]interface{}) bitbucketPushEvent {
+	// Extract push details
+	push := rawJSON["push"].(map[string]interface{})
+	changes := push["changes"].([]interface{})
+	changesObj := changes[0].(map[string]interface{})
+	new := changesObj["new"].(map[string]interface{})
+	branchName := new["name"].(string)
```

Comment:
😮 this looks wild

Reply:
The heavy nesting made it hard because I found myself making too many structs that contained only one property. If we do more heavy JSON parsing like this in the future, we can consider using a library that can abstract away those type assertions.

Examples:
https://github.com/antonholmquist/jason
https://github.com/buger/jsonparser

Reply:
I think this is way cleaner than defining the struct beforehand - let's go lib-less for now, love it!